src/iterative_ensemble_smoother/experimental_sies.py

```diff
@@ -0,0 +1,315 @@
+from __future__ import annotations
+from typing import Optional, TYPE_CHECKING, Callable
+
+import numpy as np
+import scipy as sp
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+# =============================================================================
+# MAIN CLASS
+# =============================================================================
+
+
+class SIES:
+    def __init__(
+        self,
+        param_ensemble: npt.NDArray[np.double],
+        observation_errors: npt.NDArray[np.double],
+        observation_values: npt.NDArray[np.double],
+        *,
+        seed: Optional[int] = None,
+    ):
+        self.iteration = 1
+        self.rng = np.random.default_rng(seed)
+        self.X = param_ensemble
+        self.d = observation_values
+        self.C_dd = observation_errors
+        self.A = (self.X - self.X.mean(axis=1, keepdims=True)) / np.sqrt(
+            self.X.shape[1] - 1
+        )
+
+        if self.C_dd.ndim == 2:
+            self.C_dd_cholesky = sp.linalg.cholesky(
+                self.C_dd,
+                lower=True,
+                overwrite_a=False,
+                check_finite=True,
+            )
+        else:
+            self.C_dd_cholesky = np.sqrt(self.C_dd)
+
+        # Equation (14)
+        self.D = self.d.reshape(-1, 1) + sample_mvnormal(
+            C_dd_cholesky=self.C_dd_cholesky, rng=self.rng, size=self.X.shape[1]
+        )
+
+        self.W = np.zeros(shape=(self.X.shape[1], self.X.shape[1]))
+        self.X_i = self.X
```

Comment:
This is not employed anywhere, I think, except in comments. Might be a remnant from some other code?
If this is intended to be used as `X_i = X + X@W_i  / np.sqrt(N-1)` ( adding subscript to `W`, since it is appropriate), then care must be taken to not accidentally overwrite `X` at some point. I think the original code had a `.copy()` to avoid this, which also is not that good. Probably best to avoid this one.

Reply:
Agreed. Best to avoid this one.